### PR TITLE
Refac acknowledge vault section

### PR DESCRIPTION
**Pull request by edouardparis**
base on #59

### CODE REVIEW

**Review by darosior — APPROVED** [review]
Looks good to me, but only lightly code reviewed everything but the last commit (4b00ee3d7a529e2ae29510d080d9224acb5659b9) which basically makes my entire review moot as it re-modifies most of the patches in the previous commits......

Specifically i did not look into the non-diff part of the changes, like how much logic we have relying on the VaultStatus and how the addition of tow new ones affects each part of it.

File: src/ui/state/manager.rs [hunk 1]
```diff
@@ -160,8 +160,135 @@ impl From<ManagerHomeState> for Box<dyn State> {
...
+            warning: None,
+            view: ManagerImportTransactionView::new(),
+        }
+    }
+
+    pub fn parse_pbst(&self) -> Option<Psbt> {
+        bitcoin::base64::decode(&self.psbt_input)
+            .ok()
+            .and_then(|bytes| bitcoin::consensus::encode::deserialize(&bytes).ok())
+    }
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/ui/state/manager.rs]: Issue opened with the comment

File: src/revaultd/model.rs [hunk 2]
```diff
@@ -40,8 +40,14 @@ pub enum VaultStatus {
     #[serde(rename = "funded")]
     Funded,
     /// The emergency transaction is signed
+    #[serde(rename = "securing")]
+    Securing,
+    /// The emergency transaction is signed and watchower has its payload.
```
[the remark this answers is not in the document]
Comment (edouardparis) [reply src/revaultd/model.rs:45]: Added an issue about this

File: src/ui/state/stakeholder.rs [hunk 3]
```diff
@@ -526,6 +370,27 @@ impl StakeholderDelegateFundsState {
         Command::none()
     }
 
+    pub fn on_vault_acknowledge(&mut self, outpoint: String) -> Command<Message> {
+        if let Some(selected) = &mut self.selected_vault {
+            if selected.vault.outpoint() == outpoint {
+                return selected.update(self.revaultd.clone(), VaultMessage::Acknowledge(outpoint));
+            }
+        }
```
Comment (danielabrozzoni) [inline src/ui/state/stakeholder.rs:378]: You can probably remove this piece of code as this case is already covered below, isn't it?

**Comment by danielabrozzoni** [comment]
I don't think this was introduced in this MR, but I just noticed that the stakeholder's vaults in home seem clickable, but then when clicking on them nothing happens. Is this wanted?

**Comment by edouardparis** [comment]
Yes stakeholder did not have vault selection, #65 fixes this

File: src/ui/state/stakeholder.rs (hunk 3 above)
Comment (edouardparis) [reply src/ui/state/stakeholder.rs:378]: No really, the message is passed to the selected vault if a selected vault exists (1st part) else it selects the vault and passed the message (2 part of the code)

Comment (danielabrozzoni) [reply src/ui/state/stakeholder.rs:378]: I don't see why you can't use only the second part and eventually "reselect" the same vault, but fine, it's probably more efficient anyway :)

**Review by danielabrozzoni — APPROVED** [review]
ACK 7e0cf75c4638cb364bb6f6855dda41e45e35994c

**Pull request merged**